### PR TITLE
[BUG] Skip non-converged `AutoETS` fits during model selection

### LINKED ISSUE #9291: [BUG] Maximum Likelihood optimization failed to converge
**Describe the bug**
Running AutoETS on one of Tourism competition data (series N419), gives a convergence error and the model then produces inadequate forecasts:

>Forecast: [-4.75679021e+13 -2.97761685e+10 -4.74411469e+10  1.13095811e+06
> -5.06727928e+13 -3.16493715e+10 -5.03193077e+10  1.19716407e+06]

**To Reproduce**

```python
from fcompdata import Tourism
from sktime.forecasting.ets import AutoETS
import pandas as pd
import numpy as np

ts = Tourism[419]
y_train = pd.Series(ts.x, index=pd.RangeIndex(len(ts.x)))

model = AutoETS(sp=sp, auto=True)

model.fit(y_train)
fh = np.arange(1, ts.h + 1)
forecast = model.predict(fh=fh)
```

This actually indicates the error in statsmodels:

> .../lib/python3.13/site-packages/statsmodels/base/model.py:607: ConvergenceWarning: Maximum Likelihood optimization failed to converge. Check mle_retvals

But I would suggest to catch this error and to treat it properly so that AutoETS produces robust results.

**Expected behavior**
Other implementations of ETS (R version and statsforecast) do not produce such values and do not fail.

**Additional context**

**Versions**
0.40.1 from PyPI

System:
    python: 3.13.7 (main, Jan  8 2026, 12:15:45) [GCC 15.2.0]
executable: [/home/config/Python/Libraries/smooth/python/.venv/bin/python](https://file+.vscode-resource.vscode-cdn.net/home/config/Python/Libraries/smooth/python/.venv/bin/python)
   machine: Linux-6.17.0-8-generic-x86_64-with-glibc2.42

Python dependencies:
          pip: 25.1.1
       sktime: 0.40.1
      sklearn: 1.7.2
       skbase: 0.13.1
        numpy: 2.2.6
        scipy: 1.15.3
       pandas: 2.3.3
   matplotlib: 3.10.8
       joblib: 1.5.3
huggingface-hub: None
        numba: 0.61.2
     pmdarima: None
pytorch-forecasting: None
   skforecast: None
        skpro: None
statsforecast: 2.0.3
  statsmodels: 0.14.6
 transformers: None
      tsfresh: None
      tslearn: None
        torch: 2.10.0
   tensorflow: None

### PR DESCRIPTION
**Pull request by sabasiddique1**
Treat statsmodels convergence warnings and `mle_retvals` as invalid during AutoETS model selection to avoid unstable forecasts.

#### Reference Issues/PRs
Fixes #9291

#### What does this implement/fix? Could you explain your changes?
AutoETS now ignores non‑converged statsmodels ETS fits during automatic model selection by checking convergence warnings and `mle_retvals["converged"]`. This prevents unstable models from being selected and avoids extreme forecasts.

#### What should a reviewer concentrate their feedback on?
- Convergence detection logic (warnings + `mle_retvals`)
- Model selection behavior when some fits fail to converge

### CODE REVIEW

**Comment by fkiraly** [comment]
Update: apparently, `ConvergenceWarning` has never lived in `statsmodels.base`, the import which causes the failure. Seems to be a likely AI hallucination. Adding the "AI generated" label for this reason.

**Comment by sabasiddique1** [comment]
Thanks for the suggestion, I’ve updated the import accordingly and kept the logic minimal.

I ran the ETS tests, estimator checks, and also tried a manual forecasting example locally. Everything passed, so it looks good from my side.
